lib/strings.nix

```diff
@@ -756,7 +756,14 @@ rec {
        sanitizeDerivationName pkgs.hello
        => "-nix-store-2g75chlbpxlrqn15zlby2dfh8hr9qwbk-hello-2.10"
   */
-  sanitizeDerivationName = string: lib.pipe string [
+  sanitizeDerivationName =
+  let okRegex = match "^[[:alnum:]+_?=-][[:alnum:]+._?=-]*$";
```

Comment:
You don't have to do it for performance anyway, because you'll be optmizing for the rare case.
I can see how it might improve maintainability by DRY, but that may be offset by how ugly the expression may become. You can prove me wrong in a follow-up pr, after we decide whether this whole sanitize-everything idea is good or not :)